DragonSlayer.cpp

```diff
@@ -1,23 +1,34 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
 
-//DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer (std::string name_, int hp, int armor_)
+        : Character (hp, armor_, 4), 
+          name (name_), 
+          attackItem (std::unique_ptr<AttackItem>(new AttackItem())) {}
 
-//DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so they should USE their attack item before attacking the dragon... 
         //note: they should only use the item if the dragon's hitpoints are > 0...
         //note: items are single-use only, so you need to reset it after use.  
         //look in the Character class for how the other item types are reset after use.
-        while( dragon->getHP() > 0 )
+        while ( dragon->getHP() > 0 )
         {
+            if (auto* attackItemPtr = attackItem.get())
+            {
+                attackItemPtr->use (this);
+                attackItem.reset(); //can only be used once!
+            }
```

Comment:
Do this before all of the attacking.
You are currently doing this **during** the attacking
- [ ] revise